tests/generator/generator.cpp

```diff
@@ -225,6 +225,72 @@ TEST(Generator, variableInitializedTwice)
     EXPECT_EQ(EMPTY_STRING, generator->implementationCode());
 }
 
+TEST(Generator, nonConstantInitialisingVariable)
+{
+    libcellml::ParserPtr parser = libcellml::Parser::create();
+    libcellml::ModelPtr model = parser->parseModel(fileContents("generator/non_constant_initialising_variable.cellml"));
+
+    EXPECT_EQ(size_t(0), parser->issueCount());
+
+    const std::vector<std::string> expectedIssues = {
+        "Variable 'x' in component 'main' is initialised using variable 'k2' which is not a constant.",
+    };
+    const std::vector<libcellml::Issue::Cause> expectedCauses = {
+        libcellml::Issue::Cause::GENERATOR,
+    };
+
+    libcellml::GeneratorPtr generator = libcellml::Generator::create();
+
+    generator->processModel(model);
+
+    EXPECT_EQ_ISSUES_CAUSES(expectedIssues, expectedCauses, generator);
+
+    EXPECT_EQ(libcellml::Generator::ModelType::INVALID, generator->modelType());
+
+    EXPECT_EQ(size_t(0), generator->stateCount());
+    EXPECT_EQ(size_t(0), generator->variableCount());
+
+    EXPECT_EQ(nullptr, generator->voi());
+    EXPECT_EQ(nullptr, generator->state(0));
+    EXPECT_EQ(nullptr, generator->variable(0));
+
+    EXPECT_EQ(EMPTY_STRING, generator->interfaceCode());
+    EXPECT_EQ(EMPTY_STRING, generator->implementationCode());
+}
+
+TEST(Generator, nonExistingInitialisingVariable)
+{
+    libcellml::ParserPtr parser = libcellml::Parser::create();
+    libcellml::ModelPtr model = parser->parseModel(fileContents("generator/non_existing_initialising_variable.cellml"));
+
+    EXPECT_EQ(size_t(0), parser->issueCount());
+
+    const std::vector<std::string> expectedIssues = {
+        "Variable 'x' in component 'main' is initialised using variable 'k' which cannot be found.",
+    };
```

Comment:
You've used "is not defined anywhere" earlier?  Is there a difference between defining and finding?

Reply:
No, no difference. I merely used "found" because that's what the code generator tries to do: find the variable. However, I agree that "defined" would be more suitable. Ok, done in commit 217a0c78.